### PR TITLE
[ADAM-1889][ADAM-1884] updated ReferenceRegion in python

### PR DESCRIPTION
**Pull request by akmorrow13**

### CODE REVIEW

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
Fixes https://github.com/bigdatagenomics/adam/issues/1889 and https://github.com/bigdatagenomics/adam/issues/1884

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala [hunk 1]
```diff
@@ -725,7 +725,7 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
    * @return Returns a new GenomicRDD containing only data that overlaps the
    *   querys region.
    */
-  def filterByOverlappingRegions(querys: Iterable[ReferenceRegion]): U = {
+  def filterByOverlappingRegions(querys: java.lang.Iterable[ReferenceRegion]): U = {
```
Comment (akmorrow13) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: @heuermh we needed to change this to be accessible to python (py4j takes a python iterator and converts it to java ArrayList > java.lang.Iterable). Is there a better way to do this?

**Comment by akmorrow13** [comment]
@heuermh @fnothaft ready for review. Fixes the big broken python bugs. I need some feedback on the java conversion approach for filterByOverlappingRegions()

File: adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala [hunk 2]
```diff
@@ -202,6 +202,18 @@ object ReferenceRegion {
...
 
+  /**
+   * Builds a referenceRegion from genomic coordinates.
+   *
+   * @param referenceName reference name
+   * @param start start position
+   * @param end end position
+   * @return Reference Region for these genomic coordinates
+   */
+  def fromGenomicRange(referenceName: String, start: Long, end: Long): ReferenceRegion = {
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala]: Is this method necessary?  In other words, is it difficult to call `ReferenceRegion(referenceName, start, end)` from Python?

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala (hunk 1 above)
Comment (heuermh) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: The pattern we've been using so far is to have two methods, one marked as `Java-friendly`.  I'd recommend that here, unless the Scala or Java compiler isn't smart enough to distinguish the two different versions.

File: scripts/jenkins-test [hunk 3]
```diff
@@ -214,9 +214,8 @@ then
 	    
             # we can run the python build, now that we have a spark executable
 	    mvn -U \
-		-P python,r \
-		package \
-		-DskipTests \
+		-P python \
+		test \
```
Comment (heuermh) [inline scripts/jenkins-test]: Don't we need `package` for the line `./bin/pyadam < scripts/jenkins-test-pyadam.py` below (`package` includes `test`)?

File: adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala (hunk 2 above)
Comment (akmorrow13) [reply adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala]: you cannot call ReferenceRegion.apply(referenceName, start, end) from python.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala (hunk 1 above)
Comment (akmorrow13) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: @heuermh can you give me an example of this so I can copy the style you have been using?

File: adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala (hunk 2 above)
Comment (heuermh) [reply adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala]: ok

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala (hunk 1 above)
Comment (heuermh) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: GenomicRDD has a Java friendly transmute method, https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala#L221 and others, just search for "friendly"  ;)

File: scripts/jenkins-test (hunk 3 above)
Comment (akmorrow13) [reply scripts/jenkins-test]: Doesn't seem like it, the tests are passing currently.

**Comment by akmorrow13** [comment]
ping @fnothaft for review

File: scripts/jenkins-test (hunk 3 above)
Comment (fnothaft) [reply scripts/jenkins-test]: Yeah, we shouldn't need it.

**Comment by fnothaft** [comment]
Merged! Thanks @akmorrow13!

**Pull request merged**